File: Week_00/G20200343030400/LeetCode_283_400.py
//283-move-zeros

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        count_nonzeros = 0
        for i in range(len(nums)):
            if nums[i] != 0 :
                if i > count_nonzeros :
                    nums[count_nonzeros] = nums[i]
                    nums[i] = 0
                count_nonzeros += 1 

File: Week_00/G20200343030400/LeetCode_283_400.cpp
//283-move-zeros

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int count_nonzeros = 0;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] != 0) {
                if (i != count_nonzeros) {
                    nums[count_nonzeros] = nums[i];
                    nums[i] = 0;
                }
                ++count_nonzeros;
            }
        }
    }
};
